tests/records/test_api.py

```diff
@@ -133,6 +134,35 @@ def test_draft_parent_state_hard_delete(app, db, location):
     assert RecordMetadata.query.count() == 0
 
 
+def test_new_draft_of_published_record_doesnt_override_next_draft_id(app, db, location):
+
+    # Classic draft of published record having been cleaned up scenario
+    record_1 = Record.publish(Draft.create({}))
+    db.session.commit()
+    Draft.cleanup_drafts(timedelta(0))
+
+    # Wider test for any situation where draft of record doesn't exist anymore
+    # e.g. admin force deletion
+    parent = ParentRecord.create({})
+    record_2 = Record.create({}, parent=parent)
+    record_2.register()
```

Comment:
The wider case that covers the above if cleanup_drafts works correctly and covers other scenarios e.g. direct forced deletion etc . Anything that gets us to a published record without a draft.